test/e2e/federated-service.go

```diff
@@ -260,17 +275,16 @@ func createService(fcs *federation_release_1_3.Clientset, clusterClientSets []*r
 		},
 	}
 	nservice, err := fcs.Core().Services(namespace).Create(service)
-	framework.Logf("Trying to create service %q in namespace %q", service.ObjectMeta.Name, service.ObjectMeta.Namespace)
 	Expect(err).NotTo(HaveOccurred())
 	for _, cs := range clusterClientSets {
 		waitForFederatedServiceShard(cs, namespace, nservice, 1)
 	}
 }
 
-func discoverService(f *framework.Framework, name string, exists bool) {
-	command := []string{"nslookup", name}
+func discoverService(f *framework.Framework, name string) {
+	command := []string{"sh", "-c", fmt.Sprintf("until nslookup '%s'; do sleep 1; done", name)}
```

Comment:
Agreed.  Just wanted to check.  Best check what the timeout on Eventually is.  I'm not sure offhand.